### DISCUSSION TITLE
access server config parameters?

### DISCUSSION THREAD

**Discussion by jdtsmith** [emacs-lsp/lsp-mode#3268] [Q&A] [answered]
I'm interested in accessing some of the server reported "capabilities", e.g.:

```json
    "completionProvider": {
      "workDoneProgress": true,
      "resolveProvider": true,
      "triggerCharacters": [
        ".",
        "["
      ]
    },
```

Is there a general way to access arbitrary config information the server has sent? 

Revised question: what is a reliable hook to use to access lsp--capability to make sure the server has had time to populate the capability info? initialize-hook only runs on the 1st buffer in the workspace.

**Comment by jdtsmith** [comment]
Answering my own question: `(gethash "triggerCharacters" (lsp--capability "completionProvider"))`.

**Comment by yyoncho** [comment, marked as answer]
lsp-managed-mode-hook

**Reply by jdtsmith** [reply]
Thanks, works.
